x/staking/types/validator.go

```diff
@@ -123,6 +123,33 @@ func (v Validators) Swap(i, j int) {
 	v[j] = it
 }
 
+// ValidatorsByVotingPower implements sort.Interface for []Validator based on
+// the VotingPower and Address fields.
+// Copied from tendermint/types/validator_set.go
+type ValidatorsByVotingPower []Validator
+
+func (valz ValidatorsByVotingPower) Len() int { return len(valz) }
+
+func (valz ValidatorsByVotingPower) Less(i, j int) bool {
+	if valz[i].ConsensusPower() == valz[j].ConsensusPower() {
+		var addrI, addrJ []byte
+		pkI, errI := valz[i].TmConsPubKey()
+		pkJ, errJ := valz[j].TmConsPubKey()
+		// If errors are nil, sort by address,
+		// otherwise don't sort by address and just return false
```

Comment:
is this comment correct? What happens if one of errI or errJ is non nil? will it be guaranteed to return false or could it return true if errI != nil and errJ == true?

Reply:
It will always return false, because addrI and addrJ will be `nil`

Reply:
not if one of them isn't nil. Like addI is successful buy addrJ is not

Reply:
Oh correct, just returning false in new code.